src/drop.ts

```diff
@@ -52,3 +59,17 @@ function lazyImplementation<T>(n: number): LazyEvaluator<T> {
     return { done: false, hasNext: true, next: value };
   };
 }
+
+type Drop<
+  Arr extends Array<unknown>,
+  Count extends number,
+  Removed extends Array<unknown> = [],
+> = `${Count}` extends `-${number}`
```

Comment:
**question:** can we try using the `ArraySlice` type from `type-fest`? i'd rather we reuse types from them unless we have a good reason not to

Reply:
Changed the code to use it, but seems like some cases resolve in a weird way
See all the pipe tests, and the `as number` tests